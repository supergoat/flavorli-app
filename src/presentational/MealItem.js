// @flow
import React, { Component } from "react";
import {
  Text,
  TouchableOpacity,
  View
} from "react-native";
import FAColors from "../common/FAColors";
import FAStyleSheet from "../common/FAStyleSheet";
import type { MealType } from "../common/FATypes";

/** ============================================================================
<MealItem
  meal={MealType}
  onPressMeal: (MealType) => void
/>
--------------------------------------------------------------------------------
An item in the meal list
============================================================================= */
type Props = {
  meal: MealType,
  onPressMeal: (MealType) => void
};
class MealItem extends Component<Props> {
  render() {
    const { meal, onPressMeal } = this.props;

    return (
      <TouchableOpacity
        key={meal.id}
        style={styles.meal}
        onPress={() => onPressMeal(meal)}
      >
        <Text style={styles.mealName}>{meal.name}</Text>
        <Text style={styles.mealDescription}>{meal.description}</Text>
        <Text style={styles.mealPrice}>Â£{meal.price.toFixed(2)}</Text>
      </TouchableOpacity>
    );
  }
}

/* StyleSheet
============================================================================= */
const styles = FAStyleSheet.create({
  meal: {
    borderBottomWidth: 1,
    borderBottomColor: FAColors.gallery,
    padding: 10,
    paddingVertical: 20
  },
  mealName: {
    fontSize: 18,
    color: FAColors.oxfordBlue,
    fontFamily: 'Lato'
  },
  mealDescription: {
    paddingVertical: 10,
    fontSize: 16,
    color: FAColors.osloGrey,
    fontFamily: 'Lato'
  },
  mealPrice: {
    fontSize: 16,
    color: FAColors.rollingStone,
    fontFamily: 'Lato'
  }
});

/* Export
============================================================================= */
export default MealItem;
