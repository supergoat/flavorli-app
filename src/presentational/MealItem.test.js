// @flow
import React from 'react';
import { View, Text } from 'react-native';
import MealItem from './MealItem';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

const meal = {
  id: '1',
  name: 'Lasagna',
  description: 'This is the lasagna description',
  price: 12.00
}

describe('MealItem', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <MealItem
        meal={meal}
        onPressMeal={jest.fn()}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls onPressMeal when a meal is pressed', () => {
    const onPressMealMock = jest.fn();

    const component = shallow(
      <MealItem
        meal={meal}
        onPressMeal={onPressMealMock}
      />
    );

    // Call onPress
    component.prop('onPress')();

    expect(onPressMealMock).toHaveBeenCalledWith(meal)
  });
});
