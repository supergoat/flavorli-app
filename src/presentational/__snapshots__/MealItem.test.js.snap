// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealItem renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderBottomColor": "#EEEEEE",
      "borderBottomWidth": 1,
      "opacity": 1,
      "padding": 10,
      "paddingVertical": 20,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#333C4E",
        "fontFamily": "Lato",
        "fontSize": 18,
      }
    }
  >
    Lasagna
  </Text>
  <Text
    style={
      Object {
        "color": "#7e8890",
        "fontFamily": "Lato",
        "fontSize": 16,
        "paddingVertical": 10,
      }
    }
  >
    This is the lasagna description
  </Text>
  <Text
    style={
      Object {
        "color": "#788087",
        "fontFamily": "Lato",
        "fontSize": 16,
      }
    }
  >
    Â£
    12.00
  </Text>
</View>
`;
